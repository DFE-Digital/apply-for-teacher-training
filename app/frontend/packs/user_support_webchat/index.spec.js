import userSupportWebchat from './index.js'
require('./../zendesk-stub.js')

describe('userSupportWebchat', () => {
  beforeEach(() => {
    document.body.innerHTML = `
        <div id="test-container">
          <div id="app-web-chat">
            <span class="app-web-chat__unavailable"></span>
          </div>
          <div id="launcher"></div>
        </div>
      `
    userSupportWebchat()
  })

  it('should instantiate a user support webchat', () => {
    expect(document.querySelector('#test-container')).toMatchSnapshot()
  })

  describe('hideLauncher', () => {
    it('should hide the default launcher element', () => {
      expect(document.querySelector('#launcher').classList).not.toContain('moj-hidden')
      window.setZendeskStatus('online')
      expect(document.querySelector('#launcher').classList).toContain('moj-hidden')
    })
  })
})
