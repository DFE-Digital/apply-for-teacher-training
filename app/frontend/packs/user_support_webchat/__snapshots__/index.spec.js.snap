// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userSupportWebchat should instantiate a user support webchat 1`] = `
<div
  id="test-container"
>
  
          
  <div
    id="app-web-chat"
  >
    
            
    <span
      class="app-web-chat__unavailable"
    />
    
          
    <span
      class="moj-hidden"
    >
      <a
        class="govuk-link govuk-footer__link app-web-chat__button"
        href="#"
      >
        Speak to an adviser now (opens in new window)
      </a>
    </span>
    <span
      class="app-web-chat__offline moj-hidden"
    >
      Available Monday to Friday, 10am to midday (except public holidays).
    </span>
  </div>
  
          
  <div
    id="launcher"
  />
  
        
</div>
`;
